src/components/responsive-flyout/ResponsiveFlyout.tsx

```diff
@@ -0,0 +1,48 @@
+// @flow
+import * as React from 'react';
+import classNames from 'classnames';
+// @ts-ignore flow
+import { Flyout, Overlay, OverlayHeader } from '../flyout';
+// @ts-ignore flow
+import { FlyoutProps } from '../flyout/Flyout';
+
+import './ResponsiveFlyout.scss';
+
+export interface ResponsiveFlyoutProps {
+    /** Components to render in the overlay */
+    children?: React.ReactNode;
+    /** Set className to the overlay wrapper */
+    className?: string;
+    /** Disable responsiveness behavior for higher-level components that want control */
+    isEnabled?: boolean;
+    /** Custom button to trigger for opening/closing the flyout */
+    flyoutButton: React.ReactNode;
+    /** Header for responsive layout */
+    header?: React.ReactNode;
+    /** Overlay classname */
+    overlayClassName?: string;
+}
+
+const ResponsiveFlyout = (props: ResponsiveFlyoutProps & FlyoutProps) => {
+    const flyoutRef = React.useRef<Flyout>(null);
+
+    const { header, flyoutButton, children, isEnabled = true, className, overlayClassName, ...rest } = props;
+
+    const handleCloseClick = () => {
+        if (!flyoutRef.current) return;
+
+        flyoutRef.current.closeOverlay();
+    };
+
+    return (
+        <Flyout ref={flyoutRef} className={classNames({ 'bdl-ResponsiveFlyout': isEnabled }, className)} {...rest}>
```

Comment:
It's a little odd that `isEnabled` controls this class directly, since we'd typically expect the base class to always be applied. Should we be passing `isEnabled` down to Flyout as a prop, instead?

Reply:
I wanted to enable/disable the class and have the css styles apply correspondingly. I guess I could have isEnabled pass to Flyout via isResponsive and then apply the responsive-flyout classname to it

Reply:
The other option is to just skip ResponsiveFlyout altogether

Consumers can deal with Flyout directly by passing in isResponsive and we can utilize context to pass functions to OverlayHeader like we discussed. I had a working prototype earlier

Basic structure
```
<Flyout isResponsive...>
    <Overlay ...>
        <OverlayHeader ...>
```
...
We either pass Flyout's closeOverlay or Overlay's closeOverlay handler via Context. Example below is Flyout's

In Flyout render()
```
return (
    <TetherComponent ...>
        {React.cloneElement(overlayButton, overlayButtonProps)}
        <FlyoutContext.Provider value={{closeOverlay: this.closeOverlay}}>
            {isVisible ? React.cloneElement(overlayContent, overlayProps) : null}
        </FlyoutContext.Provider>
...

```   

In OverlayHeader

```
const { closeOverlay } = React.useContext(FlyoutContext);

// closeButton onClick={closeOverlay}
```